api/rest/restapi.go

```diff
@@ -683,30 +680,30 @@ func (api *API) unpinHandler(w http.ResponseWriter, r *http.Request) {
 
 func (api *API) pinPathHandler(w http.ResponseWriter, r *http.Request) {
 	var pin types.PinSerial
-	if pinOpts := api.parsePathAndOptions(w, r); pinOpts.Path != "" {
-		logger.Debugf("rest api pinPathHandler: %s", pinOpts.Path)
+	if pinpath := api.parsePinPathOrError(w, r); pinpath.Path != "" {
+		logger.Debugf("rest api pinPathHandler: %s", pinpath.Path)
 		err := api.rpcClient.Call(
 			"",
 			"Cluster",
 			"PinPath",
-			pinOpts,
+			pinpath,
 			&pin,
 		)
 
-		api.sendResponse(w, http.StatusOK, err, &pin)
+		api.sendResponse(w, http.StatusOK, err, pin)
 		logger.Debug("rest api pinPathHandler done")
 	}
 }
 
 func (api *API) unpinPathHandler(w http.ResponseWriter, r *http.Request) {
 	var pin types.PinSerial
-	if pinOpts := api.parsePathAndOptions(w, r); pinOpts.Path != "" {
-		logger.Debugf("rest api unpinPathHandler: %s", pinOpts.Path)
+	if pinpath := api.parsePinPathOrError(w, r); pinpath.Path != "" {
+		logger.Debugf("rest api unpinPathHandler: %s", pinpath.Path)
 		err := api.rpcClient.Call(
 			"",
 			"Cluster",
 			"UnpinPath",
-			pinOpts.Path,
+			pinpath,
```

Comment:
You want to pass PinOptions for unpinning?

Reply:
No, we can pass pinpath.Path instead